velox/docs/functions/spark/string.rst

```diff
@@ -184,18 +184,21 @@ Unless specified otherwise, all functions return NULL if at least one of the arg
 
         SELECT rtrim('kr', 'spark'); -- "spa"
 
-.. spark:function:: split(string, delimiter) -> array(string)
+.. spark:function:: split(string, regex) -> array(string)
 
-    Splits ``string`` on ``delimiter`` and returns an array. ::
+    Returns an array by splitting ``string`` as many times as possible. 
+    The delimiter is any string matching regex, supported by re2.
+    This is equivalent to split(string, regex, -1), -1 is used for limit. ::
 
         SELECT split('oneAtwoBthreeC', '[ABC]'); -- ["one","two","three",""]
         SELECT split('one', ''); -- ["o", "n", "e", ""]
         SELECT split('one', '1'); -- ["one"]
 
-.. spark:function:: split(string, delimiter, limit) -> array(string)
+.. spark:function:: split(string, regex, limit) -> array(string)
    :noindex:
 
-    Splits ``string`` on ``delimiter`` and returns an array of size at most ``limit``. ::
+    Splits ``string`` on ``regex`` and returns an array of size at most ``limit``.
+    If limit is negative, ``string`` will be split as many times as possible. ::
```

Comment:
If limit is zero, the behavior is like negative. So let's clarify zero case.
See  [spark doc](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala#L529).

Reply:
Seems `split` with limit has not been supported and it will be covered by another PR: https://github.com/facebookincubator/velox/pull/8825. Maybe, we can remove the above relevant changes, and let that PR fix (if you agree, please ignore the above relevant comments).